game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/SelectMainBattleCasualtiesTest.java

```diff
@@ -0,0 +1,526 @@
+import static org.mockito.ArgumentMatchers.anyCollection;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.hamcrest.MockitoHamcrest.argThat;
+
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.DiceRoll;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.data.CasualtyDetails;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class SelectMainBattleCasualtiesTest {
+
+  @Mock IDelegateBridge delegateBridge;
+  @Mock GamePlayer gamePlayer;
+  BattleState battleState;
+
+  @BeforeEach
+  public void givenBattleState() {
+    battleState = givenBattleStateBuilder().build();
+  }
+
+  private DiceRoll givenDiceRollWithHits(final int hits) {
+    final DiceRoll diceRoll = mock(DiceRoll.class);
+    when(diceRoll.getHits()).thenReturn(hits);
+    return diceRoll;
+  }
+
+  @Test
+  void isEditMode() {
+    final List<Unit> targetUnits = List.of(givenAnyUnit(), givenAnyUnit());
+    when(battleState.getGameData().getProperties().get(TRANSPORT_CASUALTIES_RESTRICTED, false))
+        .thenReturn(false);
+    when(battleState.getGameData().getProperties().get(EDIT_MODE)).thenReturn(true);
+
+    final FiringGroup firingGroup = new FiringGroup("", "", List.of(), targetUnits);
+
+    final FireRoundState fireRoundState = new FireRoundState();
+    fireRoundState.setDice(mock(DiceRoll.class));
+
+    final SelectCasualties selectCasualties =
+        new SelectCasualties(
+            battleState,
+            BattleState.Side.OFFENSE,
+            firingGroup,
+            fireRoundState,
+            (arg1, arg2) -> new CasualtyDetails());
+
+    final SelectMainBattleCasualties.Select selectFunction =
+        mock(SelectMainBattleCasualties.Select.class);
+    final CasualtyDetails expected = new CasualtyDetails(targetUnits, List.of(), false);
+    when(selectFunction.apply(any(), any(), anyCollection(), anyInt())).thenReturn(expected);
+
+    final CasualtyDetails details =
+        new SelectMainBattleCasualties(selectFunction).apply(delegateBridge, selectCasualties);
+
+    assertThat(details.getKilled().toArray(), is(targetUnits.toArray()));
+    // edit mode always sets auto calculated to true
+    assertThat(details.getAutoCalculated(), is(true));
```

Comment:
nit, the comment on the assertion is essentially what the string arg for assertions is for : )

EG:
```
  assertThat("edit mode sets auto calcuated to true" , details.getAutoCalculated(), is(true));
```